internal/webhooks/clusterclass.go

```diff
@@ -410,6 +414,51 @@ func validateMachineHealthCheckClasses(clusterClass *clusterv1.ClusterClass) fie
 	return allErrs
 }
 
+func validateNamingStrategies(clusterClass *clusterv1.ClusterClass) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if clusterClass.Spec.ControlPlane.NamingStrategy != nil && clusterClass.Spec.ControlPlane.NamingStrategy.Template != nil {
+		if _, err := names.ControlPlaneNameGenerator(*clusterClass.Spec.ControlPlane.NamingStrategy.Template, "cluster").GenerateName(); err != nil {
+			allErrs = append(allErrs,
+				field.Invalid(
+					field.NewPath("spec", "controlPlane", "namingStrategy", "template"),
+					*clusterClass.Spec.ControlPlane.NamingStrategy.Template,
+					fmt.Sprintf("invalid template: %v", err),
+				))
+		}
+	}
+
+	for i, md := range clusterClass.Spec.Workers.MachineDeployments {
+		if md.NamingStrategy == nil || md.NamingStrategy.Template == nil {
+			continue
+		}
+		if _, err := names.MachineDeploymentNameGenerator(*md.NamingStrategy.Template, "cluster", "topology").GenerateName(); err != nil {
```

Comment:
```suggestion
		if _, err := names.MachineDeploymentNameGenerator(*md.NamingStrategy.Template, "cluster", "mdTopology").GenerateName(); err != nil {
```

maybe helps to avoid confusion just in case this somehow ends up in the reported error to users (same for MP)